### PR TITLE
New unwrap stitcher

### PR DESCRIPTION
**Pull request by mgovorcin**
Opening PR for new unwrappedPhase stitcher

### CODE REVIEW

File: tools/ARIAtools/sequential_stitching.py [hunk 1]
```diff
@@ -0,0 +1,947 @@
...
+
+from typing import List, Optional, Tuple, Union
+from osgeo import gdal, osr, gdal_array
+from pathlib import Path
+
+from matplotlib import pyplot as plt
+import matplotlib as mpl
+
+'''
+NOTES: Sequnetial stitcher relies on connected components with assumption that each component is unwrapped correctly by snaphu
```
Comment (dbekaert) [inline tools/ARIAtools/sequential_stitching.py]: small typo

File: tools/ARIAtools/sequential_stitching.py [hunk 2]
```diff
@@ -0,0 +1,947 @@
...
+        This might not always be the case, if there are unwrapping errors withing the connected component that is in the overlap
+        stitcher will not be able to compensate for this shift. Potential appraoch to correct this, would be to manually create
+        connect component around the unwrapping error and re-run stitching.
+
+        0 Component tyipcally should be masked out. Stitcher merges 0 components between two frame, if there is a lot of water
```
Comment (dbekaert) [inline tools/ARIAtools/sequential_stitching.py]: typo

File: tools/ARIAtools/sequential_stitching.py [hunk 3]
```diff
@@ -0,0 +1,947 @@
...
+
+'''
+NOTES: Sequnetial stitcher relies on connected components with assumption that each component is unwrapped correctly by snaphu
+        This might not always be the case, if there are unwrapping errors withing the connected component that is in the overlap
+        stitcher will not be able to compensate for this shift. Potential appraoch to correct this, would be to manually create
+        connect component around the unwrapping error and re-run stitching.
+
+        0 Component tyipcally should be masked out. Stitcher merges 0 components between two frame, if there is a lot of water 
+        label with 0 component, this will affect the offset correction between 0 to 0 components, which will result that that 
+        unwrapped phase within 0 components is displaced compared to other components. TODO: estimate the mean offset between
```
Comment (dbekaert) [inline tools/ARIAtools/sequential_stitching.py]: repeat

File: tools/ARIAtools/sequential_stitching.py (hunk 1 above)
Comment (dbekaert) [inline tools/ARIAtools/sequential_stitching.py]: Suggest to throw this in a quick word document to check spelling/gramar

File: tools/ARIAtools/sequential_stitching.py [hunk 4]
```diff
@@ -0,0 +1,947 @@
...
+    conncomp_data2 = conncomp_data2 + np.nanmax(conncomp_data1)
+    conncomp_data2[conncomp_data2==np.nanmax(conncomp_data1)] = 0.0
+
+    ###### GET OVERLAPS ##########
+    box_1, box_2  = frame_overlap(rdict1['SNWE'], 
+                                  rdict2['SNWE'],
+                                  [rdict1['LAT_SPACING'], rdict1['LON_SPACING']],
+                                  [rdict2['LAT_SPACING'], rdict2['LON_SPACING']])
+
+    ###### LOOP THORUGH CONNCOMPs WITHIN OVERLAP ##########
```
Comment (dbekaert) [inline tools/ARIAtools/sequential_stitching.py]: typo

**Comment by dbekaert** [comment]
@mgovorcin can you also expose the option into aria-extract code. We had the 2stage disabled at some point. So perhaps we should have the old method and your method enabled here: https://github.com/aria-tools/ARIA-tools/blob/a8585975f577a0e04c1c3cb1d1e60565df8fbcd1/tools/ARIAtools/extractProduct.py#L90

Once we are set on your method we can set the default to yours.

**Comment by dbekaert** [comment]
@mgovorcin Can you rebase on the latest and resolve conflicts?

Also in terms of integrating it in the extract class see these lines here and the comment i ade about the stitchmethod.
https://github.com/aria-tools/ARIA-tools/blob/a8585975f577a0e04c1c3cb1d1e60565df8fbcd1/tools/ARIAtools/extractProduct.py#L830-L834

**Comment by mgovorcin** [comment]
Closing this PR, and submitting changes directly to dev branch

**Pull request closed without merging**